news/2017-01-04-tip4-testing.asciidoc

```diff
@@ -15,16 +15,36 @@ tags: [tips]
 
 toc::[]
 
-In the next article of our Weld Tips series we are going to focus on integration testing of Weld applications.
-By integration testing term we mean testing at some kind of container, which in case of Weld could mean various environments.
-In each case we take advantage of link:http://arquillian.org/[Arquillian] integration testing framework.
-Let's first look at various environments and then look at link:https://github.com/arquillian/arquillian-container-weld[Arquillian Weld Embedded container], which you can use across all the environments.
+In this follow-up article of our Weld Tips series we are going to focus on integration testing of CDI/Weld applications.
+To set things straight - by integration testing we mean testing in some kind of container and hence in certain environment (SE, servlet, full EE server).
+And whatever the environment, we always make use of of link:http://arquillian.org/[Arquillian] integration testing framework. 
+
+== Arquillian as the chosen framework for testing
+While Arquillian may look complex at first, it is certainly worth it. Here are some of the advantages it gives you:
+
+* Allows usage of `@Inject` directly in test classes
+* Works well with ShrinkWrap making archive creation and deployment a breeze
+** link:http://arquillian.org/guides/shrinkwrap_introduction/[ShrinkWrap] can create any type of deployment archive (WAR, JAR, even infamous EAR) with exactly the capabilities you wish it to have
+** You can use dedicated `beans.xml` or `@Alternatives` for given test case
+* Allows to write one test and run it on any container
+** Arquillian only need correct adapter; therefore combined with Maven profiles, this gives you some nice fire power for matrix testing
+* Allows for very complex testing scenarios
+** For instance you can create several deployments which you then manually (if you so wish) deploy/undeploy
+** This makes it viable even for cluster testing (session replications for instance)
+* Can start the application server itself or can just connect to running one
+
+Should you need an inspiration, you needn't look no further than our link:https://github.com/weld/core/tree/master/tests-arquillian/src/test/java/org/jboss/weld/tests[testsuite].
+To further support the though of Arquillian usefulness for CDI/Weld testing, we may point out that even link:https://github.com/cdi-spec/cdi-tck/tree/master/impl/src/main/java/org/jboss/cdi/tck/tests[CDI TCK tests] use it.
```

Comment:
As I said I am not sure whether TCK is suitable example since there is quite a lot of extra tooling around it. Maybe better to add link to some Arquillian quickstart. WDYT?